lib/chef/provider/package/yum_tm.rb

```diff
@@ -77,23 +96,29 @@ def package_query(action, provides, arch: nil, version: nil, options: {})
           else
             logger.info("Retrieving package information from repository server (please wait)...")
 
-            cmdline = "yum info #{provides}"
-            cmdline += "-#{version} " unless version.nil?
-            cmdline += " --forcearch=#{arch} " unless arch.nil?
+            stdout = package_query_raw(provides, arch, version, options)
 
-            cmd = shell_out(cmdline)
-            if cmd.exitstatus != 0
-              raise Chef::Exceptions::Package, "#{new_resource.package_name} caused a repository error: #{cmd.stderr}"
-            else
-              Chef::Provider::Package::Yum::Version.new(
-                provides,
-                cmd.stdout.match(/^Version +: (.+)$/)[1],
-                cmd.stdout.match(/^Architecture +: (.+)$/)[1]
-              )
-            end
+            Chef::Provider::Package::Yum::Version.new(
+              provides,
+              stdout.match(/^Version +: (.+)$/)[1],
+              stdout.match(/^Architecture +: (.+)$/)[1]
+            )
           end
         end
 
+        def package_query_raw(provides, arch: nil, version: nil, options: {})
+          cmdline = "yum info #{provides}"
+          cmdline += "-#{version} " unless version.nil?
```

Comment:
I think we're missing a space here.

Reply:
at the end of line 110? Then not, as the yum syntax is `yum info cowsay` for most recent but `yum info cowsay-3.7.0` for a specific version. Or did I misunderstand?